### ISSUE TITLE
Massive ImGui input lag

### ISSUE THREAD

**Issue by duckdoom4** [vblanco20-1/vulkan-guide#118]
After following and implementing the [chapter on ImGui](https://vkguide.dev/docs/new_chapter_2/vulkan_imgui_setup/), (which I think I've done correctly since everything seems to work fine) I noticed a large input lag when I drag the ImGui window.

https://github.com/user-attachments/assets/9b8d4ac5-d917-4f1a-b5ff-4a4548fda272

As you can see in the video my FPS is stable at 60, so it's not that. 

My guess is that it's due to vsync wait order:

```
VulkanEngine::run(){
  // Update input using SDL_PollEvent/ ImGui_ImplSDL2_ProcessEvent
  // Render ImGui
 // call draw
}

VulkanEngine::draw(){
  // Wait for fence -> Already done on my GPU
  // vkAcquireNextImageKHR -> Wait for vsync
  // Wait done, so now we use the input that was collected before the wait. **Aka. last frame's data**
}
```

If I understand it correctly, the vsync wait should actually happen after present. That way we call `SDL_PollEvent` etc. right before we render the frame

**Comment by metaleap** [comment]
> If I understand it correctly, the vsync wait should actually happen after present / before SDL_PollEvent. That way we fetch the input etc. right before we render the frame

So I'm curious, did doing so fix that perf issue?

**Comment by Fabxx** [comment]
> Oh, good question. This did improve the input lag.
> 
> But IIRC, the real issue was vsync itself. I wasn't used to running apps at 60 FPS and this caused me to imagine there was more input lag then I was used to

so can you confirm that this isn't an issue with the API integration itself on imgui?

**Comment by duckdoom4** [comment]
Yes, ImGui works perfectly fine. If you run with vsync off there's no visible input lag

**Comment by Fabxx** [comment]
> Yes, ImGui works perfectly fine. If you run with vsync off there's no visible input lag

Thanks for the response, btw are you sure that whe you turn VSync on the generated FPS match the refresh rate of your monitor? If it is desync it might cause issues with the input in general, not just with the mouse.

**Comment by duckdoom4** [comment]
Yeah, my monitor has a 60hrz refresh rate. Just vsync on it's own will introduce input lag. See this comment for a good explanation: https://linustechtips.com/topic/1252057-v-sync-input-lag/?do=findComment&comment=14067963
